### DISCUSSION TITLE
OpenId Connect

### DISCUSSION THREAD

**Discussion by stelone** [strapi/strapi#7298] [Legacy Topics]
Hello, 

Can you give me more information on OpenId Connect integration ? 
Is it ready ? Pricing ? Documentation ? How to do it ? and else ? 

> For example, providing a fully featured authentication and authorization service with OAuth2 and OpenID Connect, providing an advanced Role and Permissions system, custom policies, managing 3rd party connectors like Facebook and Google.

**Comment by derrickmehaffy** [comment]
Hello @stelone currently we have not started working on the change from Grant/Purest to passport.js (which would include the options for OpenID Connect and the additional 3rd party connectors).

In the short term if the provider you want to add is on the [supported grant list](https://github.com/simov/grant#180-supported-providers--oauth-playground) you can follow [this guide](https://strapi.io/documentation/v3.x/plugins/users-permissions.html#adding-a-new-provider-to-your-project) to add the provider to your own project.
